Code/src/main/java/nl/utwente/group10/ui/components/ConnectionStateDependent.java

```diff
@@ -0,0 +1,49 @@
+package nl.utwente.group10.ui.components;
+
+import nl.utwente.group10.ui.handlers.ConnectionCreationManager;
+
+/**
+ * Interface for Nodes who are dependent on their ConnectionState.
+ */
+public interface ConnectionStateDependent {    
+    /**
+     * Tells the Node that its current state (considering connections) possibly has
+     * changed. Default implementation does not react to a potential state
+     * change.
+     *
+     * This method should only be called after the Node's constructor is done.
```

Comment:
> This method should only be called after the Node's constructor is done.

Doesn't Java guarantee this? You can't call methods on objects you haven't constructed yet.